src/lib/datepicker/calendar.ts

```diff
@@ -64,6 +77,13 @@ import {Directionality} from '@angular/cdk/bidi';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class MatCalendar<D> implements AfterContentInit, OnDestroy, OnChanges {
+
+  /** An input indicating the type of the custom header component, if set. */
+  @Input() customCalendarHeaderComponent: ComponentType<any>;
```

Comment:
lets just call the input `headerComponent` (we know its for the calendar since we're in the calendar component and the "custom" doesn't add much useful info for the user)

Reply:
done in 3467c0ac390cb4401cf7c44b79c1280ba5dcb5f8

Reply:
the input in `MatDatepicker` is still called `customCalendarHeaderComponent`. Probably this makes sense since it is exposed to the outside and should be specific..